src/main/java/net/snowflake/ingest/connection/RequestBuilder.java

```diff
@@ -132,17 +138,41 @@ private static Properties loadProperties() {
     return properties;
   }
 
-  private static String getUserAgent() {
+  /**
+   * Creates a string for user agent which should always be present in all requests to Snowflake
+   * (Snowpipe APIs)
+   *
+   * <p>Here is the format we will use:
+   *
+   * <p>SnowpipeJavaSDK/version (platform details) JAVA/<java-version>
+   *
+   * @return the default agent string
+   */
+  private static String getDefaultUserAgent() {
     final String clientVersion = PROPERTIES.getProperty("version");
+    StringBuilder defaultUserAgent = new StringBuilder(CLIENT_NAME + "/" + clientVersion);
+
+    final String osInformation =
+        String.format(
+            OS_INFO_USER_AGENT_FORMAT,
+            System.getProperty("os.name"),
+            System.getProperty("os.version"),
+            System.getProperty("os.arch"));
+
+    // append osInformation string to user agent
+    defaultUserAgent.append(" ");
+    defaultUserAgent.append(osInformation);
+    defaultUserAgent.append(" ");
+
+    // Add Java Version
     final String javaVersion = System.getProperty("java.version");
-    final String platform =
-        System.getProperty("os.name")
-            + System.getProperty("os.version")
-            + System.getProperty("os.arch");
-
-    // {client-name}/{version}/{java-version}/{platform}
-    final String userAgentFormat = "%s/%s/%s/%s";
-    return String.format(userAgentFormat, CLIENT_NAME, clientVersion, javaVersion, platform);
+    defaultUserAgent.append(JAVA_USER_AGENT + "/" + javaVersion);
+
+    return defaultUserAgent.toString();
+  }
+
+  private static String buildCustomUserAgent(String additionalUserAgentInfo) {
+    return USER_AGENT.trim() + " " + additionalUserAgentInfo;
```

Comment:
you may want to validate if `additionalUserAgentInfo` can meet the User-Agent RFC?

Reply:
The reason I didnt add was because the additional info is going to be used for monitoring purposes and there is no version and version-number.

Reply:
ok, fine. As I mentioned in the slack, just keep in mind that there is a chance that some library or validator may reject an invalid user-agent. Btw if no version exists, you could make up.

Reply:
Sounds good, I will keep this as is and once we finalize the suffix from KC side, will add checkers here.